src/BoundaryConditions.C

```diff
@@ -7,133 +7,117 @@
 // for more details.
 //
 
-
-
-#include <Realm.h>
 #include <BoundaryConditions.h>
 #include <NaluEnv.h>
 
 // yaml for parsing..
 #include <yaml-cpp/yaml.h>
 #include <NaluParsing.h>
 
-namespace sierra{
-namespace nalu{
+namespace sierra {
+namespace nalu {
 
-//==========================================================================
-// Class Definition
-//==========================================================================
-// BoundaryCondition - do some stuff
-//==========================================================================
-//--------------------------------------------------------------------------
-//-------- constructor -----------------------------------------------------
-//--------------------------------------------------------------------------
-  
-//--------------------------------------------------------------------------
-//-------- destructor ------------------------------------------------------
-//--------------------------------------------------------------------------
+// helper function for reducing code duplication in the construction process
+template <typename T>
+std::unique_ptr<BoundaryCondition>
+register_bc(const YAML::Node& node)
+{
+  std::unique_ptr<BoundaryCondition> this_bc = std::make_unique<T>();
+  auto* cast_bc = dynamic_cast<T*>(this_bc.get());
+  node >> *cast_bc;
+  return this_bc;
+}
 
-//--------------------------------------------------------------------------
-//-------- load -----------------------------------------------
-//--------------------------------------------------------------------------
+// factory method to create any supported bc
+std::unique_ptr<BoundaryCondition>
+BoundaryConditionCreator::load_single_bc_node(const YAML::Node& node)
+{
+  std::unique_ptr<BoundaryCondition> this_bc;
+  if (node["wall_boundary_condition"]) {
+    this_bc = std::move(register_bc<WallBoundaryConditionData>(node));
```

Comment:
Do you need the `move` here? register_bc returns an r-value, right?

Reply:
No, right now it returns a `std::unique_ptr` but maybe it shouldn't

Reply:
Relooking at this now and this is what I intended.  The `unique_ptr` gets passed through a couple of function calls, but ends up in the storage vector at the end.  A little messy still I guess, but functional.

Reply:
Oh wait nvm. I think you are right. I'll update it